futures-util/src/stream/mod.rs

```diff
@@ -1206,4 +1211,40 @@ pub trait StreamExt: Stream {
     fn select_next_some(&mut self) -> SelectNextSome<'_, Self> where Self: Unpin + FusedStream {
         SelectNextSome::new(self)
     }
+
+    /// Creates a future which copies all the bytes from one object to another.
+    ///
+    /// The returned future will copy all the bytes read from this `Stream` into the
+    /// `writer` specified. This future will only complete once the `stream` has hit
+    /// EOF and all bytes have been written to and flushed from the `writer`
+    /// provided.
+    ///
+    /// On success the number of bytes written is returned.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await)]
+    /// # futures::executor::block_on(async {
+    /// use futures::io::AsyncWriteExt;
+    /// use futures::stream::{self, StreamExt};
+    /// use std::io::Cursor;
+    ///
+    /// let stream1 = stream::iter(vec![b"hello".to_vec(), b"world".to_vec()]);
+    /// let mut writer = Cursor::new([0u8; 10]);
+    ///
+    /// let bytes = stream1.copy_into(&mut writer).await?;
+    /// writer.close().await?;
+    ///
+    /// assert_eq!(bytes, 10);
+    /// assert_eq!(&writer.into_inner(), b"helloworld");
+    /// # Ok::<(), Box<dyn std::error::Error>>(()) }).unwrap();
+    /// ```
+    fn copy_into<W, E>(self, writer: &mut W) -> CopyInto<'_, Self, W, E>
+    where
+        Self: Sized + Unpin,
```

Comment:
I'd include the bounds required for the resulting type to be usable as a future as well (`Self::Item: AsRef<[u8]>`)